### PR TITLE
Circle CI integration

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/33811

## Description
Adding commands for CircleCI to support commands for nightly failures dashboard.

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        (List[Dict]): List of the results.
+    """
+    results: List[Dict] = []
+    response = client_command(*client_command_args)
+    results.extend(response.get('items', []))
+    while len(results) < limit:
+        next_page_token: Optional[str] = response.get('next_page_token')
+        if not next_page_token:
+            break
+        response = client_command(*(client_command_args + [next_page_token]), next_page_token=next_page_token)
```
Comment (dantavori) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:77]: is the `next_page_token=next_page_token` required here?

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        - 'vc_type' (str): VC type. One of 'github', 'bitbucket'.
+        - 'organization' (str): Organization to retrieve workflow last runs from.
+                                Defaults to artifacts parameter is none is given.
+        - 'project' (str): Project to retrieve workflow last runs from. Defaults to project parameter is none is given.
+        - 'limit' (int): Maximum number of results to return.
+    Returns:
+        (CommandResults).
+    """
+    vc_type, organization, project, limit = get_common_arguments(client, args)
+    workflow_name: str = args.get('workflow_name', 'nightly')
```
Comment (dantavori) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:233]:
```suggestion
    workflow_name: str = args.get('workflow_name')
```
we will set the value in the relevant place (`nightly` is not necessarily a good default value for other customers)

**Review by dantavori — CHANGES_REQUESTED** [review]
great, let's arrange a demo

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py (hunk 1 above)
Comment (tomneeman151293) [reply Packs/CircleCI/Integrations/CircleCI/CircleCI.py:77]: Good question. probably not. I'll re-check this

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py (hunk 2 above)
Comment (tomneeman151293) [reply Packs/CircleCI/Integrations/CircleCI/CircleCI.py:233]: It was during my checks, i forgot to remove. my bad

**Review by dantavori — APPROVED** [review]

**Comment by shannon-holland** [comment]
@dantavori Doc review completed.

**Pull request merged**